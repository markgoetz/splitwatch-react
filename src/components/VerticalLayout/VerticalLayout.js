import React from 'react';
import PropTypes from 'prop-types';
import styles from './VerticalLayout.module.css';

export default function VerticalLayout(props) {
    let additionalClassname = '';
  
    switch (props.align) {
        case 'left':
            additionalClassname = styles.left;
            break;
        case 'right':
            additionalClassname = styles.right;
            break;
        case 'center':
            additionalClassname = styles.center;
            break;
        default:
            break;
    }

    const className = styles.verticalLayout + ' ' + additionalClassname;

    return (<div className={className}>
        {props.children}
    </div>);
}

VerticalLayout.propTypes = {
    align: PropTypes.oneOf(['left', 'center', 'right'])
};