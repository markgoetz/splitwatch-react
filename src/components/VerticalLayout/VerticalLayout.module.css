.verticalLayout {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.verticalLayout > * + * {
  margin-top: 10px;
}

@media (min-width: 769px) {
  .verticalLayout > * + * {
    margin-top: 20px;
  }
}

.center {
  align-items: center;
}

.left {
  align-items: flex-start;
}

.right {
  align-items: flex-end;
}