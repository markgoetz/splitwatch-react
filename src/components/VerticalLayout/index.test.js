import React from 'react';
import VerticalLayout from './VerticalLayout';
import { shallow } from 'preact-render-spy';

describe('Vertical Layout', () => {

  test('Vertical layout renders correctly', () => {
    const context = shallow(<VerticalLayout>
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Vertical layout renders correctly when left-aligned', () => {
    const context = shallow(<VerticalLayout align="left">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Vertical layout renders correctly when centered', () => {
    const context = shallow(<VerticalLayout align="center">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Vertical layout renders correctly when right-aligned', () => {
    const context = shallow(<VerticalLayout align="right">
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });
});