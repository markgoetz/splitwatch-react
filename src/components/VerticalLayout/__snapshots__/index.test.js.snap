// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Layout Vertical layout renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined ">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Vertical Layout Vertical layout renders correctly when centered 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Vertical Layout Vertical layout renders correctly when left-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;

exports[`Vertical Layout Vertical layout renders correctly when right-aligned 1`] = `
preact-render-spy (1 nodes)
-------
<div class="undefined undefined">
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;
